### PR TITLE
Issue 90788 improvments to System.Diagnostics.Stopwatch

### PR DESCRIPTION
**Pull request by andreas-synnerdahl**
Implementation of changes suggested in Issue 90788 to System.Diagnostics.Stopwatch to improve when measuring small time periods.

### CODE REVIEW

**Comment by andreas-synnerdahl** [comment]
@dotnet-policy-service agree

**Comment by tommcdon** [comment]
@stephentoub

File: src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs [hunk 1]
```diff
@@ -125,8 +124,19 @@ private long GetRawElapsedTicks()
...
+                if (elapsedUntilNow > 0)
+                {
+                    // When measuring small time periods the Stopwatch.Elapsed*
+                    // properties can return negative values.  This is due to
+                    // bugs in the basic input/output system (BIOS) or the hardware
+                    // abstraction layer (HAL) on machines with variable-speed CPUs
+                    // (e.g. Intel SpeedStep).
+
+                    timeElapsed += elapsedUntilNow;
+                }
```
Comment (tannergooding) [inline src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs:137]: We utilize the monotonic timers available from the underlying OS (`QueryPerformanceCounter` on Windows, `CLOCK_UPTIME_RAW` or `CLOCK_MONOTONIC` on Unix) which have strict contracts around not going backwards,

We can have up to a 1ns resolution, in which case overflow would take ~292 years in the worst possible scenario.

Given the support lifecycle of .NET and pieces of hardware, I'd say you have many other problems if you've not restarted your machine in 292 years ;)

Comment (tannergooding) [reply src/libraries/System.Private.CoreLib/src/System/Diagnostics/Stopwatch.cs:137]: The comment here (copied from the other place above) relates to much older fallback behavior that existed for very old systems before `QPC` existed (back when we supported cases like Win98 or early Win2k machines).

**Comment by stephentoub** [comment]
I think it'd be better to address https://github.com/dotnet/runtime/issues/66734 by just deleting the relevant code.

**Comment by tommcdon** [comment]
Hi @andreas-synnerdahl I've moved this PR to draft as there hasn't been much activity lately.  Feel free to move it out of draft mode if you plan on addressing the feedback.

**Pull request closed without merging**